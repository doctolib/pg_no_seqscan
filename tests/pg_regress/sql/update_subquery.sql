-- Test UPDATE with subquery

SET pg_no_seqscan.level = ERROR;

CREATE TABLE upd_foo (id bigint, value text);
CREATE TABLE upd_bar (id bigint, value text);
INSERT INTO upd_foo SELECT i, 'foo' || i FROM generate_series(1, 10) i;
INSERT INTO upd_bar SELECT i, 'bar' || i FROM generate_series(1, 10) i;

-- Test UPDATE with subquery
UPDATE upd_foo SET value = (SELECT value FROM upd_bar WHERE upd_bar.id = upd_foo.id);
