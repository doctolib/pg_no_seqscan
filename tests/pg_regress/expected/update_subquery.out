-- Test UPDATE with subquery
LOAD 'pg_no_seqscan';
SET pg_no_seqscan.level = ERROR;
CREATE TABLE upd_foo (id bigint, value text);
CREATE TABLE upd_bar (id bigint, value text);
INSERT INTO upd_foo SELECT i, 'foo' || i FROM generate_series(1, 10) i;
INSERT INTO upd_bar SELECT i, 'bar' || i FROM generate_series(1, 10) i;
-- Test UPDATE with subquery
UPDATE upd_foo SET value = (SELECT value FROM upd_bar WHERE upd_bar.id = upd_foo.id);
ERROR:  A 'Sequential Scan' on upd_foo has been detected.
  - Run an EXPLAIN on your query to check the query plan.
  - Make sure the query is compatible with the existing indexes.

Query: UPDATE upd_foo SET value = (SELECT value FROM upd_bar WHERE upd_bar.id = upd_foo.id);

-- Cleanup
DROP TABLE upd_foo, upd_bar;
